benchmarks/README.md

```diff
@@ -37,7 +37,20 @@ RobotPerf benchmarks aligns (*and will contribute*) to robotics standards so tha
 ## General concepts
 
 ### Nomenclature
-This specification follows [REP 2014](https://github.com/ros-infrastructure/rep/pull/364) nomenclature. Refer to this document for more information.
+This specification follows [REP 2014](https://github.com/ros-infrastructure/rep/pull/364) nomenclature. Refer to this document for more information. Below is the current list of all the terminology deemed important to distinguish. Recommendations for additional terminology are welcomed and can be done through pull requests or discussion during [working group meetings](https://github.com/ros-acceleration/community#meetings).
+
+- **Application.** An application refers to a particular use case that aims to achieve a specific objective in the real world. Each application will vary and may span one or more tasks or sub-tasks. An example of an application would be an autonomous vehicle, which would be associated with a specific set of tasks, sub-tasks, and workloads.
+- **Benchmark.** A benchmark is a standardized test or set of tests used to evaluate the performance of computer hardware, software, or systems. A benchmark typically consists of a workload designed to simulate real-world tasks and measure the time it takes to complete them.
+- **Benchmarking Suite.** A benchmarking suite refers to a collection of benchmarks, each typically covering one or multiple components of the robotics pipeline.
+- **Dataset.** Used synonymously with rosbags, a dataset is a tool for recording and playing back ROS message data. It is a file containing a recorded stream of ROS messages in a serialized format, along with the metadata necessary to reproduce the original message stream. The rosbag can be replayed later to simulate a specific scenario or to analyze data.
+- **Meta-package.** A meta-package is a type of package that groups together multiple related packages, providing a convenient way to manage and distribute them as a single unit.
+- **Package.** In the context of ROS2, a package is a basic unit of code organization that contains related ROS nodes, libraries, launch files, configuration files, and other resources necessary to implement a particular functionality or behavior.
+- **Reference Implementation.** A reference implementation is a standardized implementation of a benchmark that is used as a baseline for comparison with other implementations. The reference implementation serves as a common point of reference for evaluating the performance of different hardware or software configurations.
+- **Robotics Pipeline.** The robotics pipeline is a series of steps or stages that robotic systems typically go through in order to perform a specific objective. The robotics pipeline is currently comprised of five tasks: perception, localization, control, navigation, and manipulation.
+- **Task.** A task refers to a specific problem or objective a system is designed to solve or accomplish. These objectives are generally broad, such as perception or control.
```

Comment:
How does `Task` and `Category` relate to each other? Note that we're using category already in the general README, see https://github.com/robotperf/benchmarks#benchmarks.

Overall, I find a bit confusing `Task`, `Task Hierarchy` and `Sub-task` definitions. Can you elaborate on why this is necessary?

Reply:
The way I was thinking, `Category` and `Task` would be synonymous. However, I felt that the word `Category` was a bit too general as it can encompass a wide range of objectives and does not convey that a specific goal needs to be accomplished. On the other hand, the word `Task` does imply a particular action or goal.

Sub-tasks are necessary since the higher-level `Task`, such as perception, can encompass many potential objectives. There may be a better name for this, but again, it was helpful to emphasize that these are tasks to provide the implied action or goal. We included a definition of task hierarchy since the definitions we provided (i.e., task -> sub-task -> workload) produce a hierarchical representation, and this would be a helpful way to refer to this representation.

Reply:
Another reason I forgot to mention is that I wanted to keep the terminology consistent with that of MLPerf, which uses the terms `task` and `workload` in their [paper](https://arxiv.org/pdf/1911.02549.pdf).